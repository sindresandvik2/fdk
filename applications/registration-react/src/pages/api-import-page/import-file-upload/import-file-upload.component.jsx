import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'reactstrap';
import _ from 'lodash';

import localization from '../../../utils/localization';
import { PanelBase } from '../../../components/panel-base/panel-base.component';
import { InputFile } from '../../../components/input-file/input-file.component';
import { AlertMessage } from '../../../components/alert-message/alert-message.component';

export const ImportFileUpload = props => {
  const {
    fileName,
    handleFileUpload,
    handleCancelImport,
    handleShowLinkImport
  } = props;
  return (
    <PanelBase>
      <div className="mb-5">
        <h3>{localization.api.import.importApiSpecFile}</h3>
      </div>
      <div className="mb-5">
        <AlertMessage type="danger">
          {localization.api.import.importApiSpecFileError}
          <button
            className="fdk-button-small fdk-color-blue-dark"
            onClick={handleShowLinkImport}
          >
            {localization.api.import.toImportLink}
          </button>
        </AlertMessage>
      </div>
      <div className="mb-3">
        <span>{localization.api.import.filename}: </span>
        <span>{fileName}</span>
      </div>
      <InputFile
        className="btn btn-primary mr-3 mb-0"
        onChange={handleFileUpload}
      >
        {localization.api.import.importNewFile}
      </InputFile>
      <Button
        className="fdk-color-blue-dark fdk-bg-color-white"
        color="transparent"
        onClick={handleCancelImport}
      >
        {localization.api.import.cancel}
      </Button>
    </PanelBase>
  );
};

ImportFileUpload.defaultProps = {
  fileName: null,
  handleShowLinkImport: _.noop()
};

ImportFileUpload.propTypes = {
  fileName: PropTypes.string,
  handleFileUpload: PropTypes.func.isRequired,
  handleCancelImport: PropTypes.func.isRequired,
  handleShowLinkImport: PropTypes.func
};
