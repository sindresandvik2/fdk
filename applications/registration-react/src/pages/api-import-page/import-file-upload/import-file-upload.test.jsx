import React from 'react';
import { shallow } from 'enzyme';
import { ImportFileUpload } from './import-file-upload.component';

let defaultProps;
let wrapper;
let preventDefault;
let handleFileUpload;
let handleCancelImport;

beforeEach(() => {
  handleFileUpload = jest.fn();
  handleCancelImport = jest.fn();
  preventDefault = jest.fn();
  defaultProps = {
    fileName: 'Api.json',
    handleFileUpload,
    handleCancelImport
  };
  wrapper = shallow(<ImportFileUpload {...defaultProps} />);
});

test('should render ImportFileUpload correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle input file upload', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.find('InputFile').prop('onChange')(mockedEvent);
  expect(handleFileUpload).toHaveBeenCalled();
});

test('should handle button cancel', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.find('Button').prop('onClick')(mockedEvent);
  expect(handleCancelImport).toHaveBeenCalled();
});
