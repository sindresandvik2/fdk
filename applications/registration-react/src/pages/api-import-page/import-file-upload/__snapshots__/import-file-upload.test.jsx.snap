// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ImportFileUpload correctly 1`] = `
<PanelBase>
  <div
    className="mb-5"
  >
    <h3>
      Importer API-spesifikasjon via fil
    </h3>
  </div>
  <div
    className="mb-5"
  >
    <AlertMessage
      type="danger"
    >
      Noe gikk galt med importen. Kontroller at du har valgt riktig fil, eller prøv å 
      <button
        className="fdk-button-small fdk-color-blue-dark"
      >
        importere via lenke
      </button>
    </AlertMessage>
  </div>
  <div
    className="mb-3"
  >
    <span>
      Filnavn
      : 
    </span>
    <span>
      Api.json
    </span>
  </div>
  <InputFile
    className="btn btn-primary mr-3 mb-0"
    onChange={[MockFunction]}
  >
    Importer ny fil
  </InputFile>
  <Button
    className="fdk-color-blue-dark fdk-bg-color-white"
    color="transparent"
    onClick={[MockFunction]}
    tag="button"
  >
    Avbryt
  </Button>
</PanelBase>
`;
